doc/source/dev/contributor/debugging_linalg_issues.rst

```diff
@@ -0,0 +1,440 @@
+for more details.
+
+
+FlexiBLAS
+`````````
+
+`FlexiBLAS <https://github.com/mpimd-csc/flexiblas>`__ provides runtime
+switching support (among other things) for all installed BLAS libraries that it
+can detect. There are a few limitations at the time of writing (March 2024),
+primarily: no support for Windows, no support for macOS Accelerate (the updated
+version, with ``NEWLAPACK`` symbols). If those limitations don't matter for
+you, FlexiBLAS can be a quite useful tool for efficient debugging, including
+for versions of OpenBlAS and other BLAS libraries that you have to build from
+source.
+
+Once you have everything set up, the development experience is::
+
+    $ python dev.py build -C-Dblas=flexiblas -C-Dlapack=flexiblas
+    $ FLEXIBLAS=NETLIB python dev.py test -s linalg
+    $ FLEXIBLAS=OpenBLAS python dev.py test -s linalg
+    # Or export the environment variable to make the selection stick:
+    $ export FLEXIBLAS=OpenBLAS
+
+You can also provide a path to a built BLAS library (e.g.,
+``FLEXIBLAS="libbhlas_atlas.so"``) - see the `usage docs in its README
+<https://github.com/mpimd-csc/flexiblas#selecting-the-backend-at-runtime>`__
+for more details.
+
+Unless you're on Fedora, you will likely have to build FlexiBLAS from source,
+which is a bit of work. The good news is that this should work no matter if
+you're on Linux or macOS, and use Python via virtualenvs, conda environments,
+or in some other way. We'll go through how to build OpenBLAS and FlexiBLAS
+from source, to allow debugging whether something in the latest OpenBLAS
+version is different from Netlib BLAS/LAPACK (or MKL) or not.
+
+The below should work in any environment where you can build SciPy itself; the
+only additional tool we need is CMake (install with, for example, ``pip install
+cmake``).
+
+Clone each repository::
+
+    $ cd ..  # starting from the root of the local `scipy` repo
+    $ mkdir flexiblas-setup && cd flexiblas-setup
+    $ git clone https://github.com/OpenMathLib/OpenBLAS.git openblas
+    $ git clone https://github.com/mpimd-csc/flexiblas.git
+    $ mkdir built-libs  # our local prefix to install everything to
+
+Build OpenBLAS::
+
+    $ cd openblas
+    $ mkdir build && cd build
+    $ cmake .. -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=$PWD/../../built-libs
+    $ cmake --build . -j
+    $ cmake --install . --prefix $PWD/../../built-libs
+    $ cd ../..
+
+Build FlexiBLAS::
+
+    $ mkdir build && cd build
+    $ # Note: this will also pick up the libraries in your system/env libdir
+    $ cmake .. -DEXTRA="OpenBLAS" -DLAPACK_API_VERSION=3.9.0 \
+        -DOpenBLAS_LIBRARY=$PWD/../../built-libs/lib/libopenblas.so \
+        -DCMAKE_INSTALL_PREFIX=$PWD/../../built-libs
+    $ cmake --build . -j
+    $ cmake --install . --prefix $PWD/../../built-libs
+    $ cd ../..
+
+We're now ready to build SciPy against FlexiBLAS::
+
+    $ cd scipy
+    $ python dev.py build -C-Dblas=flexiblas -C-Dlapack=flexiblas
```

Comment:
```
$ python dev.py build -C-Dblas=flexiblas -C-Dlapack=flexiblas
💻  meson setup /home/ev-br/repos/scipy/build --prefix /home/ev-br/repos/scipy/build-install 
-Dblas=flexiblas -Dlapack=flexiblas
The Meson build system
Version: 1.3.2
Source dir: /home/ev-br/repos/scipy
Build dir: /home/ev-br/repos/scipy/build
Build type: native build
Project name: scipy
Project version: 1.13.0.dev0
C compiler for the host machine: /home/ev-br/.conda/envs/scipy-dev/bin/x86_64-conda-linux-gnu-cc (gcc 12.3.0 "x86_64-conda-linux-gnu-cc (conda-forge gcc 12.3.0-5) 12.3.0")
C linker for the host machine: /home/ev-br/.conda/envs/scipy-dev/bin/x86_64-conda-linux-gnu-cc ld.bfd 2.40
C++ compiler for the host machine: /home/ev-br/.conda/envs/scipy-dev/bin/x86_64-conda-linux-gnu-c++ (gcc 12.3.0 "x86_64-conda-linux-gnu-c++ (conda-forge gcc 12.3.0-5) 12.3.0")
C++ linker for the host machine: /home/ev-br/.conda/envs/scipy-dev/bin/x86_64-conda-linux-gnu-c++ ld.bfd 2.40
Cython compiler for the host machine: cython (cython 3.0.9)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python3 found: YES (/home/ev-br/.conda/envs/scipy-dev/bin/python3.11)
Found pkg-config: YES (/home/ev-br/.conda/envs/scipy-dev/bin/pkg-config) 0.29.2
Run-time dependency python found: YES 3.11
Program cython found: YES (/home/ev-br/.conda/envs/scipy-dev/bin/cython)
Compiler for C supports arguments -Wno-unused-but-set-variable: YES 
Compiler for C supports arguments -Wno-unused-function: YES 
Compiler for C supports arguments -Wno-conversion: YES 
Compiler for C supports arguments -Wno-misleading-indentation: YES 
Library m found: YES
Fortran compiler for the host machine: /home/ev-br/.conda/envs/scipy-dev/bin/x86_64-conda-linux-gnu-gfortran (gcc 12.3.0 "GNU Fortran (conda-forge gcc 12.3.0-5) 12.3.0")
Fortran linker for the host machine: /home/ev-br/.conda/envs/scipy-dev/bin/x86_64-conda-linux-gnu-gfortran ld.bfd 2.40
Compiler for Fortran supports arguments -Wno-conversion: YES 
Checking if "-Wl,--version-script" : links: YES 
Program tools/generate_f2pymod.py found: YES (/home/ev-br/.conda/envs/scipy-dev/bin/python3.11 /home/ev-br/repos/scipy/tools/generate_f2pymod.py)
Program scipy/_build_utils/tempita.py found: YES (/home/ev-br/.conda/envs/scipy-dev/bin/python3.11 /home/ev-br/repos/scipy/scipy/_build_utils/tempita.py)
Program pythran found: YES 0.15.0 0.15.0 (/home/ev-br/.conda/envs/scipy-dev/bin/pythran)
Found CMake: /home/ev-br/.conda/envs/scipy-dev/bin/cmake (3.29.0)
WARNING: CMake Toolchain: Failed to determine CMake compilers state
Run-time dependency xsimd found: NO (tried pkgconfig and cmake)
Run-time dependency threads found: YES
Library npymath found: YES
Library npyrandom found: YES
Run-time dependency pybind11 found: YES 2.11.1
Run-time dependency flexiblas found: NO (tried pkgconfig and cmake)

scipy/meson.build:165:9: ERROR: Dependency "flexiblas" not found, tried pkgconfig and cmake

A full log can be found at /home/ev-br/repos/scipy/build/meson-logs/meson-log.txt
Meson build setup failed!

```

It seems to not know it needs to look into `../flexiblas-setup/built-libs/`?

Reply:
This worked:

```
$ export PKG_CONFIG_PATH=/home/ev-br/repos/flexiblas-setup/built-libs/lib/pkgconfig/
$ python dev.py build -C-Dblas=flexiblas -C-Dlapack=flexiblas
...
```

the incantation is technically present below in the "creating reproducers" section (which is where I took it). Would be great to duplicate it here though.